packages/data-stores/src/site/queries/use-site-media-storage.ts

```diff
@@ -0,0 +1,36 @@
+import { useQuery, type UseQueryResult } from '@tanstack/react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+import useQueryKeysFactory from './lib/use-query-keys-factory';
+import type { SiteMediaStorage, RawSiteMediaStorage } from '../types';
+
+interface Props {
+	siteIdOrSlug: string | number | null | undefined;
+}
+
+/**
+ * Site media storage from `/sites/[ siteIdOrSlug ]/media-storage` endpoint
+ * transposed to camelCase format
+ */
+function useSiteMediaStorage( {
+	siteIdOrSlug,
+}: Props ): UseQueryResult< SiteMediaStorage | undefined > {
+	const queryKeys = useQueryKeysFactory();
+
+	return useQuery( {
+		queryKey: queryKeys.siteMediaStorage( siteIdOrSlug ),
+		queryFn: async (): Promise< SiteMediaStorage | undefined > => {
+			const mediaStorage: RawSiteMediaStorage = await wpcomRequest( {
```

Comment:
It looks like `wpcomRequest` is a generic, so rather than typing the response we can let TS infer it:

`const mediaStorage = await wpcomRequest< RawSiteMediaStorage >( { ...`

Reply:
thanks. sounds good :-)